morphir/core/src/org/finos/morphir/ir/writers.scala

```diff
@@ -46,9 +85,27 @@ trait IRTypeWriters extends NamingWriters { self: Annotator =>
     }
   }
 
-  implicit def TypeWriter[A: Writer]: Writer[Type[A]] = ???
+  implicit def TypeWriter[A: Writer]: Writer[Type[A]] = new Writer[Type[A]] {
+    override def write0[R](out: Visitor[_, R], v: Type[A]): R = v match {
+      case r: T.Type.ExtensibleRecord[A] => ExtensibleRecordTypeWriter.write0(out, r)
+      case r: T.Type.Function[A]         => FunctionTypeWriter.write0(out, r)
+      case r: T.Type.Record[A]           => RecordTypeWriter.write0(out, r)
+      case r: T.Type.Reference[A]        => ReferenceTypeWriter.write0(out, r)
+      case r: T.Type.Tuple[A]            => TupleTypeWriter.write0(out, r)
+      case r: T.Type.Unit[A]             => UnitTypeWriter.write0(out, r)
+      case r: T.Type.Variable[A]         => VariableTypeWriter.write0(out, r)
+    }
+  }
 
-  implicit def FieldTypeWriter[A: Writer]: Writer[T.Field[A]] = ???
+  implicit def FieldTypeWriter[A: Writer]: Writer[T.Field[A]] = new Writer[T.Field[A]] {
```

Comment:
With a field we should get output that conforms to this:

https://github.com/finos/morphir-elm/blob/792dca4158a779617bfda40051a0810dc02c38e8/src/Morphir/IR/Type/Codec.elm#L149-L154

Reply:
fixed